--- man-db-2.6.1/configure.ac.recompress~	2012-02-14 13:12:00.000000000 +0100
+++ man-db-2.6.1/configure.ac	2012-04-15 11:03:45.685510358 +0200
@@ -346,6 +346,8 @@ MAN_CHECK_PROGS([xz], [XZ], [use XZ as L
 if test -n "$xz"
 then
 	unxz="$xz -dc"
+	compressor="xz -c9e"
+	compress_ext="xz"
 fi
 MAN_CHECK_PROGS([lzip], [LZIP], [use LZIP as Lempel-Ziv-Markov chain-Algorithm compression utility], [lzip])
 if test -n "$lzip"
