diff -up man-db-2.12.1/init/systemd/man-db.conf.in.2~ man-db-2.12.1/init/systemd/man-db.conf.in
--- man-db-2.12.1/init/systemd/man-db.conf.in.2~	2024-07-09 01:28:14.662662350 +0200
+++ man-db-2.12.1/init/systemd/man-db.conf.in	2024-07-09 01:28:44.842986863 +0200
@@ -1,3 +1,3 @@
-d /var/cache/man 0755 @cache_top_owner@ @cache_top_owner@ a:1w
+d /var/cache/man 0755 root root a:1w
 x /var/cache/man/index*
 x /var/cache/man/*/index*
