diff -up man-db-2.9.0/configure.ac.omv~ man-db-2.9.0/configure.ac
--- man-db-2.9.0/configure.ac.omv~	2020-01-05 02:10:53.013898465 +0100
+++ man-db-2.9.0/configure.ac	2020-01-05 02:12:40.688893241 +0100
@@ -333,6 +333,11 @@ if test -n "$gzip" || test -n "$compress
 then
 	AC_DEFINE([COMP_CAT], [1], [Define if you have compressors and want to support compressed cat files.])
 fi
+MAN_CHECK_PROGS([zstd], [ZSTD], [use ZSTD as zstd compression utility], [zstd])
+if test -n "$zstd"
+then
+	unzstd="$zstd -dc"
+fi
 AC_SUBST([compressor])
 AC_SUBST([compress_ext])
 AC_SUBST([gunzip])
@@ -341,6 +346,7 @@ AC_SUBST([bunzip2])
 AC_SUBST([unlzma])
 AC_SUBST([unxz])
 AC_SUBST([unlzip])
+AC_SUBST([unzstd])
 MAN_COMPRESS_LIB([z], [gzopen])
 dnl To add more decompressors just follow the scheme above.
 
diff -up man-db-2.9.0/include/comp_src.h.in.omv~ man-db-2.9.0/include/comp_src.h.in
--- man-db-2.9.0/include/comp_src.h.in.omv~	2020-01-05 02:08:28.039905499 +0100
+++ man-db-2.9.0/include/comp_src.h.in	2020-01-05 02:10:06.821900706 +0100
@@ -77,6 +77,12 @@ struct compression comp_list[] = {
 	{UNLZIP, "lz", NULL},
 #endif /* HAVE_LZIP */
 
+/* If we have zstd, incorporate the following */
+#ifdef HAVE_ZSTD
+	{UNZSTD, "zst", NULL},
+	{UNZSTD, "zstd", NULL},
+#endif /* HAVE_ZSTD */
+
 /*------------------------------------------------------*/
 /* Add your decompressor(s) and extension(s) below here */
 /*------------------------------------------------------*/
diff -up man-db-2.9.0/include/manconfig.h.in.omv~ man-db-2.9.0/include/manconfig.h.in
--- man-db-2.9.0/include/manconfig.h.in.omv~	2020-01-05 02:07:52.758907210 +0100
+++ man-db-2.9.0/include/manconfig.h.in	2020-01-05 02:08:12.339906260 +0100
@@ -225,6 +225,7 @@
 #define UNLZMA "@unlzma@"
 #define UNXZ "@unxz@"
 #define UNLZIP "@unlzip@"
+#define UNZSTD "@unzstd@"
 
 /*-----------------------------------------------------------------------*/
 /* The things below here shouldn't really be changed unless you really	 */
